src/surge-xt/gui/SurgeGUIEditor.cpp

```diff
@@ -4733,6 +4737,73 @@ juce::PopupMenu SurgeGUIEditor::makeMidiMenu(const juce::Point<int> &where)
     return midiSubMenu;
 }
 
+#ifdef SURGE_HAS_OSC
+juce::PopupMenu SurgeGUIEditor::makeOSCMenu(const juce::Point<int> &where)
+{
+    auto oscSubMenu = juce::PopupMenu();
+    int portnum = juceEditor->processor.oscListener.portnum;
+
+    oscSubMenu.addItem(Surge::GUI::toOSCase("List all parameters"),
+            [this]() {
+                showHTML(this->parametersToHtml()); 
+            });
+
+    oscSubMenu.addSeparator();
+    oscSubMenu
+        .addItem(
+            Surge::GUI::toOSCase("Change OSC port (" + std::to_string(portnum) + ")"),
+            [this, portnum]() {
+                const auto c{std::to_string(portnum)};
+                promptForMiniEdit(
+                    c, "Enter a new value:", "OSC Port #", juce::Point<int>(10, 10),
+                    [this](const std::string &c) {
+                        int newPort = ::atoi(c.c_str());
+                        juceEditor->processor.changeOSCPort(newPort);
+                        Surge::Storage::updateUserDefaultValue(&(synth->storage),
+                                                               Surge::Storage::OSCPort, newPort);
+                    },
+                    mainMenu);
+            });
+
+    if (portnum != DEFAULT_OSC_PORT)
+    {
+        oscSubMenu.addItem(
+            Surge::GUI::toOSCase("Reset OSC port (to " + std::to_string(DEFAULT_OSC_PORT) + ")"),
+            [this]() {
+                juceEditor->processor.changeOSCPort(DEFAULT_OSC_PORT);
+                Surge::Storage::updateUserDefaultValue(&(synth->storage), Surge::Storage::OSCPort,
+                                                       DEFAULT_OSC_PORT);
+            });
+    }
+
+    bool alreadyListening = juceEditor->processor.oscListener.listening;
+    if (alreadyListening) {
+         oscSubMenu.addItem(Surge::GUI::toOSCase("Stop OSC listener"),
+            [this]() {
+                juceEditor->processor.oscListener.stopListening();
+            });
+   } else {
+         oscSubMenu.addItem(Surge::GUI::toOSCase("Start OSC listener"), [this]() {
+             int defaultOSCPort = Surge::Storage::getUserDefaultValue(
+                 &(this->synth->storage), Surge::Storage::OSCPort, DEFAULT_OSC_PORT);
+             std::cout << "default osc port: " << defaultOSCPort;
```

Comment:
This looks like a diagnostic message and should be removed.

Reply:
Now wrapped with #ifdef DEBUG